include/aikido/control/ros/RosJointStateClient.hpp

```diff
@@ -0,0 +1,68 @@
+#ifndef AIKIDO_CONTROL_ROS_ROSJOINTSTATECLIENT_HPP_
+#define AIKIDO_CONTROL_ROS_ROSJOINTSTATECLIENT_HPP_
+#include <string>
+#include <boost/circular_buffer.hpp>
+#include <dart/dynamics/dynamics.hpp>
+#include <ros/ros.h>
+#include <ros/callback_queue.h>
+#include <sensor_msgs/JointState.h>
+
+namespace aikido {
+namespace control {
+namespace ros {
+
+class RosJointStateClient
+{
+public:
+  /// Constructor.
+  /// \param _skeleton Skeleton to read JointState updates for.
+  /// \param _nodeHandle ROS node.
+  /// \param _topicName Name of topic to subscribe to for JointState updates.
+  RosJointStateClient(
+    dart::dynamics::SkeletonPtr _skeleton,
+    ::ros::NodeHandle _nodeHandle,
+    const std::string& _topicName,
+    size_t capacity);
+
+  /// Call all callbacks in mCallbackQueue.
+  void spin();
+
+  /// Returns the last position of each joint.
+  /// \param _metaSkeleton Skeleton to read DOFs from.
+  Eigen::VectorXd getLatestPosition(
+    const dart::dynamics::MetaSkeleton& _metaSkeleton) const;
+
+  // TODO: implement
+  // getPositionAtTime(const MetaSkeleton&, const ros::Time&, bool)
```

Comment:
This is useful for two reasons
- If multiple nodes are publishing `JointState` messages on the same topic, then the elements of `getLatestPosition()` may have been populated at different times. This can lead to some strange artifacts, e.g. where the end-effector pose oscillates if you execute a synchronized arm + base trajectory.
- If you want to synchronize the configuration of the robot to an external sensor (e.g. a camera or RGBD sensor), there may be considerable latency between the sensor capturing data and the data reaching this process. This is problematic when implementing algorithms like a [self filter](http://wiki.ros.org/robot_self_filter#Robot_Self_Filtering).